hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -140,18 +140,14 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
    */
   protected lazy val (tableAvroSchema: Schema, internalSchemaOpt: Option[InternalSchema]) = {
     val schemaResolver = new TableSchemaResolver(metaClient)
-    val internalSchemaOpt = if (!isSchemaEvolutionEnabled) {
-      None
-    } else {
-      Try {
-        specifiedQueryTimestamp.map(schemaResolver.getTableInternalSchemaFromCommitMetadata)
-          .getOrElse(schemaResolver.getTableInternalSchemaFromCommitMetadata)
-      } match {
-        case Success(internalSchemaOpt) => toScalaOption(internalSchemaOpt)
-        case Failure(e) =>
-          logWarning("Failed to fetch internal-schema from the table", e)
-          None
-      }
+    val internalSchemaOpt = Try {
+      specifiedQueryTimestamp.map(schemaResolver.getTableInternalSchemaFromCommitMetadata)
```

Comment:
@fuyun2024 Can you please create a JIRA for this change? This does not look like a minor change.

Here, commit metadata will always be scanned. Previously, it would be scanned only when schema evolution was enabled.

Reply:
@fuyun2024 unfortunately we can't do that -- @nsivabalan is actually currently working on the opposite change that will make sure that Schema Evolution is guarded behind the flag such that we can force it disabled if needs to be (even though it might not be safe for some use-cases, there are cases when this still is necessary)